gcc/rust/resolve/rust-late-name-resolver-2.0.cc

```diff
@@ -262,5 +262,36 @@ Late::visit (AST::StructExprStructFields &s)
   DefaultResolver::visit (s);
 }
 
+void
+Late::visit (AST::GenericArgs &args)
+{
+  for (auto &lifetime : args.get_lifetime_args ())
+    visit (lifetime);
+
+  for (auto &generic : args.get_generic_args ())
+    visit (generic);
+
+  for (auto &binding : args.get_binding_args ())
+    visit (binding);
+}
```

Comment:
Doesn't the default visitor already visit generic args ?

Reply:
Kind of, from what I can tell `GenericArg::accept_vis` just visits the properties of `GenericArg` and there isn't a virtual method for `visit (GenericArg &)`. Therefore, in order to get the `GenericArgs` visitor to call into the non-virtual `GenericArg` visitor, the former has to be overridden along with the latter.

Reply:
I've rebased and added a comment to the `GenericArgs` visitor method.